kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt

```diff
@@ -320,13 +356,12 @@ class TypeCollector(private val provider: AstProvider, private val options: Opti
                         provider.error("@Provides method must have a concrete implementation", method)
                         continue
                     } else {
-                        providesMethods.add(method)
+                        providesMethods.add(method to methodScope)
                     }
                 } else if (method.isProvider()) {
-                    val scope = method.scopeType(options)
-                    if (scope != null) {
+                    if (methodScopeTypes != null) {
```

Comment:
should this be `isNotEmpty()`?

Reply:
This case is different because it is pulling it out of the `scopeTypes` Map on line 314

Reply:
It's a little confusing because the invariant is that if there is a `Set<AstType>` in `scopeTypes` then it won't be empty.

Reply:
ah looking at the full context I understand